src/llama-sampling.cpp

```diff
@@ -1637,7 +1639,7 @@ static struct llama_sampler * llama_sampler_init_grammar_impl(
                 trigger_pattern += std::regex_replace(trigger_words[i], special_chars, "\\$0");
             }
             trigger_pattern += ")[\\s\\S]*";
-            const auto * trigger_pattern_c = trigger_pattern.c_str();
+            trigger_pattern_c = trigger_pattern.c_str();
             trigger_patterns = &trigger_pattern_c;
```

Comment:
This is still a dangling pointer.

Reply:
@marek-hradil Regardless of where the `trigger_pattern` variable is created within this function, it will be released when the `llama_sampler_init_grammar_impl` function finishes execution, which means `trigger_pattern_c` will still be left dangling.

Reply:
@o7si Thanks, you are fully correct. It is not **actually** a dangling pointer. :+1: 

Your change proposal reads good. I would not have used a `std::vector<>` as a temporary though. Either use `std::array<T, 1>` or just a c array (of const char pointers).

Reply:
@o7si Hi, perfect! I'll do that.